### ISSUE TITLE
Label should only be used to provide the most basic description of the input purpose

### ISSUE THREAD

**Issue by ghost** [cis-bathnes/pattern-lib#6]
![image](https://user-images.githubusercontent.com/609519/34080015-b8eb8f70-e32f-11e7-95b8-8a522db2c18f.png)
![image](https://user-images.githubusercontent.com/609519/34080116-fffd1c8e-e330-11e7-8ea8-b939b417f1ce.png)

Extra information such as hint text and error messages ideally shouldn't be contained in the label Instead this should be outside the label but associated using `aria-describedby`.

```
<div class="form-group form-group-error">
   <label for="example-full-name" id="error-full-name">
      <span class="form-label-bold">Full name</span>
   </label>
   <span class="form-hint">As shown on your birth certificate or passport</span>
   <span class="error-message">Error message about full name goes here</span>
   <input class="form-control form-control-error" id="example-full-name" type="text" name="fullName" value="" aria-describedby="form-hint error-message">
</div>
```

Notes:

- `aria-describedby` can reference multiple element IDs 
- reference to the error message must be removed when the error is not present this includes removing the whole `aria-describedby` attribute when no other elements are referenced (hint text is not present)

**Comment by ghost** [comment]
![image](https://user-images.githubusercontent.com/609519/34080151-5941dc80-e331-11e7-9d00-205f8003a1c5.png)

**Comment by BenDarby** [comment]
Moved hint text and error message outside of label across toolkit and added aria-describedby where needed.
